### PR TITLE
fix: execute user-defined commands on start/exit nodes

### LINKED ISSUE #69: Start & Exit nodes are not being executed
In such DIP file:

```
workflow TestShowcase
  goal: "Test node tools"
  start: Start
  exit: Exit

  tool Start
    label: "Start"
    timeout: 10s
    command:
      set -eu
      echo "start" >start.txt
      printf 'start'
  
  tool Exit
    label: "Exit"
    timeout: 10s
    command:
      set -eu
      echo "exit" >exit.txt
      printf 'exit'
  
  tool Tool
    label: "Test tool"
    timeout: 10s
    command:
      set -eu
      echo "foo" >tool.txt
      printf 'bar'
  
  edges
    Start -> Tool
    Tool -> Exit
```

The only visible artifact is `tool.txt`, the `start.txt` and `exit.txt` do not materialize. Apparently this is because those steps are executed using `start`/`exit` handlers.

So, how to encode start/exit nodes properly? There is no `start`/`exit` node type defined in Dippin? Just add dummy tool call like the above, and ignore it? It works, but doesn't seem to be the right way.

### PR DESCRIPTION
**Pull request by clintecker**
## Summary

- **Root cause**: `ensureStartExitNodes` in `pipeline/dippin_adapter_edges.go` checked only for a `prompt` attribute to determine if a start/exit node had meaningful content. Tool nodes use `tool_command` and human nodes use `mode` — neither has a `prompt` — so their handlers were silently overwritten with the passthrough `start`/`exit` no-ops.
- **Fix**: Replace the `prompt`-only check with a `nodeHasHandlerContent()` helper that recognises all three handler-specific attrs (`prompt`, `tool_command`, `mode`). Bare placeholder nodes with none of these still get the passthrough. All other node types keep their real handler.
- **Backward compat**: Bare agent nodes with no prompt (used as lightweight start/exit placeholders) still get the passthrough handler, preserving existing behavior.

## Test plan

- [ ] `TestEnsureStartExitNodes_ToolNodeKeepsHandler` — unit: tool start/exit nodes keep `"tool"` handler
- [ ] `TestEnsureStartExitNodes_HumanNodeKeepsHandler` — unit: human start/exit nodes keep `"wait.human"` handler
- [ ] `TestFromDippinIR_ToolStartExitNodes` — adapter-level: full `FromDippinIR` round-trip with three tool nodes (Start, Middle, Exit) all get `"tool"` handler
- [ ] `TestNodeHasHandlerContent` — helper: table-driven tests for all attr combinations
- [ ] All 15 existing packages still pass (`go test ./... -short`)
- [ ] Race detector clean (`go test -race ./pipeline/...`)
- [ ] `dippin doctor` — A grade on all three core pipelines

Closes #69

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Start/exit node handlers are now preserved for non-bare agent nodes (tools, human-mode, parallel, conditional, subgraph, etc.); only bare agent/codergen nodes with no prompt receive passthrough start/exit handlers.

* **Tests**
  * Added regression tests covering handler-preservation scenarios and edge cases.

* **Documentation**
  * Updated changelog and guidance to reflect the refined handler decision rules.

### CODE REVIEW

**Comment by clintecker** [comment]
@copilot please address all the review feedback

**Comment by clintecker** [comment]
Fixed in 79a650a. `nodeHasHandlerContent` now inspects `n.Handler` rather than sniffing attrs — immune to `AgentConfig.Params` keys "mode"/"tool_command" leaking (Codex P2). Added `TestEnsureStartExitNodes_ParamsLeakPrevention` for the regression. Integration tests for parallel/fan-in/conditional/subgraph as start/exit also added (Copilot #3088722132, #3088722188).

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- **Start/exit node handler overwrite broadened fix**: `ensureStartExitNodes` previously checked only three specific attributes (`prompt`, `tool_command`, `mode`) to decide whether to preserve a node's handler. Nodes using other handler types (`parallel` with `parallel_targets`, `parallel.fan_in` with `fan_in_sources`, `conditional`, `subgraph`, `stack.manager_loop`, or `wait.human` with default mode) would still have their handlers silently overwritten. The helper now bases the decision on the resolved `Handler` field: any handler other than `codergen` is always preserved; only a bare `codergen` node with no `prompt` gets the passthrough. Closes #69.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply CHANGELOG.md]: Fixed in 6806c83.

File: pipeline/dippin_adapter_edges.go [hunk 2]
```diff
@@ -106,7 +106,11 @@ func synthesizeFanInEdges(g *Graph, irNode *ir.Node, cfg ir.FanInConfig, existin
 
 // ensureStartExitNodes verifies that the start and exit nodes exist in the graph.
 // The start/exit shapes (Mdiamond/Msquare) are always set so the validator can
-// identify them. Nodes without a prompt also get the passthrough handler.
+// identify them. Only bare codergen (agent) nodes with no prompt are treated as
+// passthrough placeholders and get the start/exit handler. Nodes with any other
+// resolved handler (tool, wait.human, parallel, parallel.fan_in, conditional,
```
[the remark this answers is not in the document]
Comment (clintecker) [reply pipeline/dippin_adapter_edges.go:111]: Fixed in 6806c83.

File: pipeline/dippin_adapter_test.go [hunk 3]
```diff
@@ -1399,3 +1399,235 @@ func TestConvertEdge_ZeroWeightOmitted(t *testing.T) {
...
+	}
+	if graph.Nodes["Exit"].Attrs["tool_command"] != "touch exit.txt" {
+		t.Errorf("Exit tool_command = %q, want touch exit.txt", graph.Nodes["Exit"].Attrs["tool_command"])
+	}
+}
+
+// TestNodeHasHandlerContent verifies the helper that distinguishes bare nodes
+// from nodes with handler-specific content. The helper is based on the resolved
+// Handler field: any handler other than "codergen" is considered meaningful.
+// A codergen node with no prompt is the only passthrough case.
```
[the remark this answers is not in the document]
Comment (clintecker) [reply pipeline/dippin_adapter_test.go]: Fixed in 6806c83.
